preprocess_text.py

```diff
@@ -0,0 +1,72 @@
+import nltk
+import gensim
+from nltk.tokenize import word_tokenize
+import numpy as np
+import pandas as pd
+
+def tokenize_a_doc(s):
+    try:
+        tok = word_tokenize(' '.join(s.split(" ")).lower())
+    except:
+        tok = ""
+    return tok
+
+def vectorize_text(wordlist,wdic):
+    return [wdic.get(word) if wdic.get(word) is not None else 0 for word in wordlist]
+    
+'''
+# Arguments
+    x: numpy array of text data that wants to be modelled
+    name: just a fancy name used to save word2vec embeddings etc
+    word2vec: If None, new word2vec will be generated when calling word2vec_init.
+        If a path, it will try to load existing a word2vec file (nb: custom format)
+    word_dim: Dimensionality of word2vec vectors if new ones are trained.
+'''
+
+class Preprocess_text:
+    def __init__(self,x,name="textdata",word2vec_path=None, word_dim=50):
+        # Download nltk data if it doesnt exist
+        nltk.download('punkt')
+        self.name = name
+        self.df = pd.DataFrame(x,columns=["text"])
+        self.word2vec_path = word2vec_path
+        self.word_dim = word_dim
+        
+    def tokenize_text(self):
+        self.df['tokenized'] = self.df.text.map(tokenize_a_doc)
+        
+    def word2vec_init(self,word_dim = 50):
+        if self.word2vec_path is None:
+            # Create word2vecs from text data
+            sentences = self.df.tokenized.values.tolist()
+            print 'Training on %d documents.' %(len(sentences))
+            
+            # Build word2vec model
+            gensimmodel = gensim.models.word2vec.Word2Vec(sentences, size=self.word_dim, window=5, min_count=5, workers=15)
```

Comment:
Are these 15 workers for a spark-context of some sort? Should it be an input to the method?